File: DataAccess/Models/Product/ProductModel.cs
namespace ECommerce.DataAccess.Models.Product
{
    public class ProductModel
    {
        public int ProductId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int ManufacturerId { get; set; }
        public int CategoryId { get; set; }

        public ProductModel(string name, string description, decimal price, int manufacturerId, int categoryId)
        {
            ProductId = -1;
            Name = name;
            Description = description;
            Price = price;
            ManufacturerId = manufacturerId;
            CategoryId = categoryId;
        }

        public ProductModel(int productId, string name, string description, decimal price, int manufacturerId, int categoryId)
        {
            ProductId = productId;
            Name = name;
            Description = description;
            Price = price;
            ManufacturerId = manufacturerId;
            CategoryId = categoryId;
        }
    }
}

File: frontend/src/app/models/product-model.ts
import CategoryModel from './category-model';
import ManufacturerModel from './manufacturer-model';
import StoreModel from './store-model';

export default interface ProductModel {
  productId: number;
  name: string;
  description: string;
  price: number;
  manufacturerId: number;
  categoryId: number;
  manufacturer: ManufacturerModel;
  category: CategoryModel;
  store: StoreModel | null;
  imageUrls: string[];
}
